src/lazy/preload/utils.ts

```diff
@@ -17,4 +19,11 @@ export function insertLinkTag(href: string, rel: string) {
   link.href = href;
 
   document.head?.appendChild(link);
+
+  return () => {
+    // Remove the child if it is still in the document head
+    if (document.head?.contains(link)) {
```

Comment:
We could do `link.parentNode === document.head` that is much faster that `contains`

Reply:
Good thinking!